edxval/models.py

```diff
@@ -286,6 +286,13 @@ def create_or_update(cls, course_video, file_name, image_data=None):
         """
         video_image, created = cls.objects.get_or_create(course_video=course_video)
         if image_data:
+            # Delete the existing image only if this image is not used by anyone else. This is necessary because
+            # after a course re-run, a video in original course and the new course points to same image, So when
+            # we update an image in new course and delete the existing image. This will delete the image from
+            # original course also leaving video with no image.
```

Comment:
last line can be made a bit better : `original course as well, thus leaving video with having no image`